libvast/vast/system/node_state.hpp

```diff
@@ -0,0 +1,24 @@
+#ifndef VAST_SYSTEM_NODE_STATE_HPP
+#define VAST_SYSTEM_NODE_STATE_HPP
+
+#include "vast/expression.hpp"
+#include "vast/filesystem.hpp"
+
+#include "vast/system/tracker.hpp"
+
+namespace vast {
+namespace system {
+
+/// A container for VAST components.
+struct node_state {
+  path dir;
+  tracker_type tracker;
+  std::unordered_map<std::string, int> labels;
+  std::string name = "node";
+  std::unordered_map<expression, caf::actor> continuous_queries;
```

Comment:
I think the TRACKER is better suited for tracking this relationship. The TRACKER already maintains all running components, which is not the job of NODE.

Other than being simpler, did you have a particular reason why NODE is the right place?